backend/dropbox/dropbox.go

```diff
@@ -1586,6 +1617,18 @@ func (o *Object) Open(ctx context.Context, options ...fs.OpenOption) (in io.Read
 		return
 	}
 
+	if o.non_downloadable {
+		arg := files.ExportArg{Path: o.id, ExportFormat: o.export_format}
+		err = o.fs.pacer.Call(func() (bool, error) {
+			var exportResult *files.ExportResult
+			exportResult, in, err = o.fs.srv.Export(&arg)
+			o.bytes = int64(exportResult.ExportMetadata.Size)
```

Comment:
These should go after the `o.fs.pacer.Call` when you are sure it has suceeded otherwise you'll certainly case a `nil` pointer exception.